Add September Office Hours Blog

burtonr opened the pull request:
This commit also adds Burton as an author

Signed-off-by: Burton Rheutan <rheutan7@gmail.com>

Resolves #30 

This isn't complete as I would like some input from @stefanprodan for some more information about the Istio integration.

Also, I need to listen through the roundtable to complete the community section. I will do this first thing US morning

burtonr commented:
@alexellis This is nearly complete. Just need to add a bit about the Istio integration and update the date for tomorrow. Let me know if what you think so far.

@stefanprodan are you able to add some information to the Istio section? Or should I fill it out based on what you mention in the demo/call?

stefanprodan commented:
@burtonr yes I've added the info on the issue #30

alexellis commented on _posts/2018-09-12-september-office-hours.md:
```diff
@@ -0,0 +1,72 @@
+* Bart Smykla (VMware)
+  * Recent full time OpenFaaS team member.
+  * Working on support for authentication gateway with Kafka connector with annotations
+* Mark Pundsack (GitLab)
```
Remove please.

alexellis commented on _posts/2018-09-12-september-office-hours.md:
```diff
@@ -0,0 +1,72 @@
+## Istio Demonstration
+_needing input from Stefan_
+
+## Community Roundtable
```
Here is a summary of updates from the community.

alexellis commented on _posts/2018-09-12-september-office-hours.md:
```diff
@@ -0,0 +1,72 @@
+  * Looking for a comparison with kNative
+  * OpenFaaS goal is to be easy to use and production ready, where kNative is still an alpha project with a lot of potential
+* Radoslav Dimitrov
+  * Supporting ARM compatibility. Adding support for ARM64.
```
ARMHF compatibility updated in all repos.

alexellis commented on _posts/2018-09-12-september-office-hours.md:
```diff
@@ -0,0 +1,72 @@
+  * Article available [here](https://opensparkl.org/docker/example/openfaas/orchestration/saga/2018/09/07/openfaas/)
+
+## Get Involved
+Give OpenFaaS a try with our self-paced [workshop](https://docs.openfaas.com/tutorials/workshop/)
```
Give x a try -> Try OpenFaaS today with

alexellis commented on _posts/2018-09-12-september-office-hours.md:
```diff
@@ -0,0 +1,72 @@
+
+## What's New
+* [Stateless Microservices]({% post_url 2018-09-06-stateless-microservices %})
+  * Deploy a regular microservice as if it were a function
```
Deploy serverless microservices with OpenFaaS and manage them just like functions.

alexellis replied:
Also add "Kubernetes"

alexellis commented on _posts/2018-09-12-september-office-hours.md:
```diff
@@ -0,0 +1,72 @@
+> We also maintain an [official channel](https://www.youtube.com/channel/UCdKi97g5FmzvrmtIp9FyOVA) that contains demos from some of the meetings
+
+## Project Goals and Purpose
+OpenFaaS is focused on developers first, easy for operators (_not_ a headache), and the great community aspect
```
(_not_ a headache) - good colour, but omit on this occasion.

alexellis commented on _posts/2018-09-12-september-office-hours.md:
```diff
@@ -0,0 +1,72 @@
+## Project Goals and Purpose
+OpenFaaS is focused on developers first, easy for operators (_not_ a headache), and the great community aspect
+
+* Developer first
```
In the call I mentioned some additional work we did to support this like the `faas-cli up` - we can link to the blog post?

alexellis commented on _posts/2018-09-12-september-office-hours.md:
```diff
@@ -0,0 +1,72 @@
+dark_background: true
+---
+
+The OpenFaaS community holds regular office hours for anyone to join in as well as specialized discussions with project members, and core contributors. During these discussions, we talk about what's new in the project, where we're headed, and any other updates individual members may have.
```
Mention the three types of calls and their purpose (covered in video) and roughly how many people that corresponds to.

johnmccabe replied:
I commented elsewhere on that, I think that could be better in its own post about community and what it takes to run/build/maintain a vibrant project.

alexellis commented on _posts/2018-09-12-september-office-hours.md:
```diff
@@ -0,0 +1,72 @@
+
+The OpenFaaS community holds regular office hours for anyone to join in as well as specialized discussions with project members, and core contributors. During these discussions, we talk about what's new in the project, where we're headed, and any other updates individual members may have.
+
+All of the office hours meetings are recorded and available for later viewing on YouTube.
```
Link to the playlist here.

burtonr replied:
I wasn't able to find an actual playlist of the meetings recordings. They're all uploaded to your YouTube account, but not stored in a specific playlist. 

Should they be moved to the OpenFaaS channel?

alexellis commented on _posts/2018-09-12-september-office-hours.md:
```diff
@@ -0,0 +1,72 @@
+
+<iframe width="700" height="394" src="https://www.youtube.com/embed/gkKpercV6FA?rel=0" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe>
+
+> We also maintain an [official channel](https://www.youtube.com/channel/UCdKi97g5FmzvrmtIp9FyOVA) that contains demos from some of the meetings
```
Instead of: We maintain an official channel.

Subscribe to the Official OpenFaaS YouTube Channel where you can view the highlights from our meetings including demos of the newest features we're working on.

alexellis commented on _posts/2018-09-12-september-office-hours.md:
```diff
@@ -0,0 +1,72 @@
+author_staff_member: burton
+dark_background: true
+---
+
```
True (keep it in as a generic statement) but highlight the content of this particular call's content. This is the text that you should imagine being in a Twitter card etc.

alexellis commented on _posts/2018-09-12-september-office-hours.md:
```diff
@@ -0,0 +1,72 @@
+* [Growing list of end users](https://docs.openfaas.com/#users-of-openfaas)
+  * Add your company [here](https://github.com/openfaas/faas/issues/776)
+* [OpenFaaS Cloud](https://docs.openfaas.com/openfaas-cloud/intro/)
+  * GitOps for your functions
```
What was it re-defined as in the call?

alexellis commented on _posts/2018-09-12-september-office-hours.md:
```diff
@@ -0,0 +1,72 @@
+## What's New
+* [Stateless Microservices]({% post_url 2018-09-06-stateless-microservices %})
+  * Deploy a regular microservice as if it were a function
+* [Growing list of end users](https://docs.openfaas.com/#users-of-openfaas)
```
Why does this matter? What does it show?

Traction. Growing traction

burtonr commented:
@alexellis and @rgee0  I missed your comments earlier (I assume it was due to the Github outage), but I've addressed them now.

I added a section about the discussion on kNative as I thought it was a really great question and response to include. However, the community updates section of the video may be edited out to focus on the project updates. If so, I'll remove that piece. 

@rgee0 I don't think this style of post really gets a begin/middle/end as it's more of a recap. I do like the thought of more quarterly, or seasonal, updates. Maybe we could do that for the end of October/beginning of November as the end of the season. In that case, should we do a post for every community call and favor the broader seasonal updates?

johnmccabe commented on _posts/2018-09-13-september-office-hours.md:
```diff
@@ -0,0 +1,87 @@
+description: Multi-user OpenFaaS Cloud, Istio Service Mesh, and Stateless Microservices
+
+date: 2018-09-13
+image: /images/september-office-hours/team.jpg
```
@alexellis should we restrict images to those with provable sources wrt licensing, CC/Public Domain etc?

burtonr replied:
I'm not sure how to _prove_ that the image is free for use, but here's the link to where I downloaded it from: https://pixabay.com/en/team-team-building-success-computer-3373638/

johnmccabe replied:
In this case showing the source has a CCA licence is the proof. The alternative is just random Google images sourced images where copyright is not clear.

burtonr replied:
Is there somewhere that should be stored? In the post, or commit message?

johnmccabe replied:
I think commit/PR message is a very good idea. Worth updating the contributing doc while you've this open to make future posters aware.

johnmccabe commented on _posts/2018-09-13-september-office-hours.md:
```diff
@@ -0,0 +1,87 @@
+dark_background: true
+---
+
+This month's OpenFaaS office hours meeting talks about OpenFaaS Cloud as a multi-user platform for teams, how we've improved the ability to run stateless microservices managed as functions, and includes a demo of Istio service mesh integration.
```
Replace "This month's" with "September's"

johnmccabe replied:
"talks about" -> "saw presentations from the community of new features", with a bullet list:
- Multi-user support in OpenFaas Cloud
- Support for Stateless Microservices in OpenFaaS (and link to the blog post)
- Istio Service Mesh integration.

johnmccabe commented on _posts/2018-09-13-september-office-hours.md:
```diff
@@ -0,0 +1,87 @@
+
+> All of the office hours meetings are recorded and available for later viewing on YouTube.
+
+<iframe width="700" height="394" src="https://www.youtube.com/embed/IQCnHMn1wdI?rel=0" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe>
```
Use the youtube include tag for this thats been used in other posts (have a look at the 2018-07-14-kubernetes-operator-crd.md for example).

burtonr replied:
Very cool! Didn't realize there was a special tag/parameter. Updated

johnmccabe commented on _posts/2018-09-13-september-office-hours.md:
```diff
@@ -0,0 +1,87 @@
+* Great Community
+  * 130+ contributors
+  * Most have multiple commits, coming back helping build the project
+  * Built by the community, for you to use
```
This feels out of place with the post description - warrants a seperate blog post about project goals perhaps.

I'd potentially move all of the sections about the community meetings to the. And I'd group them under a header.

The title/description of the post is about describing what was demoed/discussed, not describing that a community meeting is, hence pushing that to the end as a secondary section. (It potentially warrants being a post of its own perhaps - a broader one on how a community driven project organises and disseminates information/feedback etc)

burtonr replied:
This was included as it was a speaking point in the talk. It's not a main point of the meeting, but was brought up. As this is a recap of the meeting, I thought it would be appropriate to include.

I do agree that we should write something up highlighting the various meetings, their purpose, and how they are successful.

johnmccabe commented on _posts/2018-09-13-september-office-hours.md:
```diff
@@ -0,0 +1,87 @@
+  * Add your company [here](https://github.com/openfaas/faas/issues/776)
+* [OpenFaaS Cloud](https://docs.openfaas.com/openfaas-cloud/intro/)
+  * GitOps for your functions
+  * Becoming a platform to run OpenFaaS for multiple users or a team
```
Feel like the post as a whole would be more in keeping with other posts with a prose based approach rather than the bullet list of points, wdyt?

(they would be super useful in the YT video description with timestamps for sure - always wished more people did that)

burtonr replied:
I wasn't sure how to write up a recap without it being either a bullet list of points made, or just a transcript or the recording. 

My goal here was to provide enough information to satisfy the curious without making it a lot to digest. If someone wanted more information they could either view the video, or dive into the linked resource. 

I considered including links with the timestamps, but thought it didn't really fit in a blog post. As you said, that should be in the video description with the timestamp links (plus it's much easier to do there)

johnmccabe commented:
I do 1000000% think this is a **great** idea going forward however, and we should end up with effectively a template for what is in effect a round up post - and a bit of an advertisement of the progress and evolution of the project.

burtonr commented:
@johnmccabe Thanks for the feedback! I've made the updates.

We should discuss something of a template for these types of posts as we will be doing a lot of them, assuming each meeting gets a post (which I think it should).

rgee0 commented on _posts/2018-09-13-september-office-hours.md:
```diff
@@ -0,0 +1,87 @@
+## Istio Integration
+
+"I wanted to integrate OpenFaaS with Istio to take advantage of Istio builtin security features and tracing capabilities. Some of the features demoed are: mutual TLS between OpenFaaS core services and functions, Jaeger tracing for function calls, function isolation with Mixer policies." - _Stefan Prodan_
+
```
Embed https://youtu.be/iO7PYGhRro0 here?

burtonr commented:
With the comments above, I'm going to close this in favour of a broader "Autum Update" post that will allow a more traditional beginning/middle/end and have more content to warrant a full post. 

It's difficult to create an entire post for a single community chat, however, having a quarterly, or seasonal update would provide some sense of the journey the project is making.
